packages/components/src/date-time/time.js

```diff
@@ -122,43 +191,113 @@ class TimePicker extends Component {
+				'is-12-hour': is12Hour,
+				'is-24-hour': ! is12Hour,
+			} ) }>
+				<fieldset>
+					<legend className="components-datetime__time-legend invisible">{ __( 'Date' ) }</legend>
+					<div className="components-datetime__time-wrapper">
+						<div className="components-datetime__time-field components-datetime__time-field-month">
+							<select
+								aria-label={ __( 'Month' ) }
+								className="components-datetime__time-field-month-select"
+								value={ month }
+								onChange={ this.onChangeMonth }
+								onBlur={ this.updateMonth }
+							>
+								<option value="01">{ __( 'January' ) }</option>
+								<option value="02">{ __( 'February' ) }</option>
+								<option value="03">{ __( 'March' ) }</option>
+								<option value="04">{ __( 'April' ) }</option>
+								<option value="05">{ __( 'May' ) }</option>
+								<option value="06">{ __( 'June' ) }</option>
+								<option value="07">{ __( 'July' ) }</option>
+								<option value="08">{ __( 'August' ) }</option>
+								<option value="09">{ __( 'September' ) }</option>
+								<option value="10">{ __( 'October' ) }</option>
+								<option value="11">{ __( 'November' ) }</option>
+								<option value="12">{ __( 'December' ) }</option>
+							</select>
+						</div>
+						<div className="components-datetime__time-field components-datetime__time-field-day">
+							<input
+								aria-label={ __( 'Day' ) }
+								className="components-datetime__time-field-day-input"
+								type="number"
+								value={ day }
+								step={ 1 }
+								min={ 1 }
+								onChange={ this.onChangeDay }
+								onBlur={ this.updateDay }
+							/>
+						</div>
+						<div className="components-datetime__time-field components-datetime__time-field-year">
+							<input
+								aria-label={ __( 'Year' ) }
+								className="components-datetime__time-field-year-input"
+								type="number"
+								step={ 1 }
+								value={ year }
+								onChange={ this.onChangeYear }
+								onBlur={ this.updateYear }
+							/>
+						</div>
+					</div>
+				</fieldset>
+
+				<fieldset>
+					<legend className="components-datetime__time-legend invisible">{ __( 'Time' ) }</legend>
+					<div className="components-datetime__time-wrapper">
+						<div className="components-datetime__time-field components-datetime__time-field-time">
+							<input
+								aria-label={ __( 'Hours' ) }
+								className="components-datetime__time-field-hours-input"
+								type="number"
+								step={ 1 }
+								min={ this.getMinHours() }
+								max={ this.getMaxHours() }
+								value={ hours }
+								onChange={ this.onChangeHours }
+								onBlur={ this.updateHours }
+							/>
+							<span
+								className="components-datetime__time-separator"
+								aria-hidden="true">:</span>
+							<input
+								aria-label={ __( 'Minutes' ) }
+								className="components-datetime__time-field-minutes-input"
+								type="number"
+								value={ minutes }
```

Comment:
For these inputs, we don't validate until blur, meaning the user can enter a value like `80` for minutes. Not a huge issue, but we could constrain with `min` and `max`.